### PR TITLE
Doc/subsvc validation

### PR DESCRIPTION
**Pull request by ioangut**
## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] Make sure `mvn install` runs for the whole project and, if you touched any code in the respective service, submission and distribution service can be run with `spring-boot:run`

## Description

Documentation for Submission Service - Validation

### CODE REVIEW

File: docs/SUBMISSION.md [hunk 1]
```diff
@@ -18,3 +18,17 @@ Profile                                           | Effect
 `disable-ssl-client-verification-verify-hostname` | Disables the verification of the SSL hostname for the connection to the verification server.
 
 Please refer to the inline comments in the base `application.yaml` configuration file for further details on the configuration properties impacted by the above profiles.
+
+## Submission Validations
+
+Custom Annotation `@ValidSubmissionPayload`
```
Comment (MKusber) [inline docs/SUBMISSION.md:24]: Please use markdown headers, i.e. `###`

Comment (MKusber) [inline docs/SUBMISSION.md:24]:
```suggestion
### Custom Annotation `@ValidSubmissionPayload`
```

File: docs/SUBMISSION.md [hunk 2]
```diff
@@ -18,3 +18,17 @@ Profile                                           | Effect
...
+
+You will find the implementation file at [`/services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java`](/services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java)
+
+Validation Constraints:
```
Comment (MKusber) [inline docs/SUBMISSION.md:28]:
```suggestion
### Validation Constraints:
```

File: docs/SUBMISSION.md [hunk 3]
```diff
@@ -18,3 +18,17 @@ Profile                                           | Effect
...
+
+   StartIntervalNumber values from the same `SubmissionPayload` shall be unique(should not receive the same key twice in the same package).
```
Comment (MKusber) [inline docs/SUBMISSION.md:30]:
```suggestion
* `StartIntervalNumber` values from the same `SubmissionPayload` shall be unique.
```

File: docs/SUBMISSION.md [hunk 4]
```diff
@@ -18,3 +18,17 @@ Profile                                           | Effect
...
+
+   There must not be any keys in the `SubmissionPayload` that have overlapping time windows.
+
+   The period covered by the data file must not exceed the configured maximum number of days( limit of 14 days/package). No submission with more than 14 keys.
```
Comment (MKusber) [inline docs/SUBMISSION.md:34]:
```suggestion
* There must not be any keys in the `SubmissionPayload` that have overlapping time windows.
* The period covered by the data file must not exceed the configured maximum number of days (limit of 14 days/package). No submission with more than 14 keys.
```

File: docs/SUBMISSION.md [hunk 5]
```diff
@@ -18,3 +18,15 @@ Profile                                           | Effect
 `disable-ssl-client-verification-verify-hostname` | Disables the verification of the SSL hostname for the connection to the verification server.
 
 Please refer to the inline comments in the base `application.yaml` configuration file for further details on the configuration properties impacted by the above profiles.
+
+## Submission Validations
+
+### Custom Annotation `@ValidSubmissionPayload`
```
Comment (stevesap) [inline docs/SUBMISSION.md:28]:
```suggestion
### Custom Annotation [`@ValidSubmissionPayload`](https://corona-warn-app.github.io/cwa-server/1.0.0/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.html)
```

File: docs/SUBMISSION.md [hunk 6]
```diff
@@ -18,3 +18,15 @@ Profile                                           | Effect
...
+
+You will find the implementation file at [`/services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java`](/services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java)
+
+### Validation Constraints
+
+* `StartIntervalNumber` values from the same `SubmissionPayload` shall be unique.
```
Comment (stevesap) [inline docs/SUBMISSION.md:34]:
```suggestion
* `StartIntervalNumber` values from the same [`SubmissionPayload`](https://corona-warn-app.github.io/cwa-server/1.0.0/app/coronawarn/server/common/protocols/internal/SubmissionPayload.html) shall be unique.
```

File: docs/SUBMISSION.md [hunk 7]
```diff
@@ -18,3 +18,15 @@ Profile                                           | Effect
...
+* There must not be any keys in the `SubmissionPayload` that have overlapping time windows.
```
Comment (stevesap) [inline docs/SUBMISSION.md:35]:
```suggestion
* There must not be any keys in the [`SubmissionPayload`](https://corona-warn-app.github.io/cwa-server/1.0.0/app/coronawarn/server/common/protocols/internal/SubmissionPayload.html) that have overlapping time windows.
```

File: docs/SUBMISSION.md [hunk 8]
```diff
@@ -25,4 +25,12 @@ Coming soon
...
-Coming soon
+### Custom Annotation `@ValidSubmissionPayload`
+
+You will find the implementation file at [`/services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java`](/services/submission/src/main/java/app/coronawarn/server/services/submission/validation/ValidSubmissionPayload.java)
+
+### Validation Constraints
+
+* `StartIntervalNumber` values from the same `SubmissionPayload` shall be unique.
+* There must not be any keys in the `SubmissionPayload` that have overlapping time windows.
+* The period covered by the data file must not exceed the configured maximum number of days (limit of 14 days/package). No submission with more than 14 keys.
```
Comment (stevesap) [inline docs/SUBMISSION.md:36]:
```suggestion
* The period covered by the data file must not exceed the configured maximum number of days, which is defined by `max-number-of-keys` property in [`application.yaml`](/services/submission/src/main/resources/application.yaml). Currently no submissions with more than 14 keys are accepted

**Pull request merged**